docs/reference/run.md

```diff
@@ -154,14 +154,15 @@ The operator can identify a container in three ways:
 -   UUID short identifier ("f78375b1c487")
 -   Name ("evil_ptolemy")
 
-The UUID identifiers come from the Docker daemon, and if you do not
-assign a name to the container with `--name` then the daemon will also
-generate a random string name too. The name can become a handy way to
-add meaning to a container since you can use this name when defining
-[*links*](../userguide/dockerlinks.md) (or any
-other place you need to identify a container). This works for both
-background and foreground Docker containers.
-
+The UUID identifiers come from the Docker daemon. If you do not assign a
+container name with the `--name` option, then the daemon generates a random
+string name for you. Defining a `name` can be a handy way to add meaning to a
+container. If you specify a `name`, you can use it  when referencing the
+container within a Docker network. This works for both background and foreground
+Docker containers.
+
+**Note**: Service discovery is unavailable on the default bridge network.
```

Comment:
I dont think we need to call out this. But IMO the following sentence explaining the need to have `links` for containers to communicate by name for `default bridge network` is more appropriate.